File: app/src/main/java/com/prism/poc/PrismTrackerApplication.java
package com.prism.poc;

import android.app.Application;

/**
 * Created by Krishna Upadhya on 19/03/20.
 */
public class PrismTrackerApplication extends Application {
    private static PrismTrackerApplication instance;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }

    public static PrismTrackerApplication getInstance() {
        return instance;
    }

}

File: app/src/main/java/com/prism/poc/PrismTrackerApplication.kt
package com.prism.poc

import android.app.Application
import com.prism.poc.db.AppDatabase
import com.prism.poc.db.LocationRepository

/**
 * Created by Krishna Upadhya on 19/03/20.
 */
class PrismTrackerApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    companion object {
        @JvmStatic
        var instance: PrismTrackerApplication? = null
            private set
    }

    val database by lazy { AppDatabase.getDatabase(this) }
    val repository by lazy { LocationRepository(database.locationDao()) }
}